File: Sample-java/src/main/java/com/opera/ads/demo/fragment/RewardedFragment.java
package com.opera.ads.demo.fragment;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.opera.ads.AdError;
import com.opera.ads.AdFormat;
import com.opera.ads.demo.R;
import com.opera.ads.RewardItem;
import com.opera.ads.rewarded.RewardedAd;
import com.opera.ads.rewarded.RewardedAdInteractionListener;
import com.opera.ads.rewarded.RewardedAdLoadListener;

public class RewardedFragment extends BaseFragment {

    @Nullable
    private RewardedAd mRewardedAd;

    @Override
    protected boolean hasVideo() {
        return false;
    }

    @NonNull
    @Override
    protected AdFormat getAdFormat() {
        return AdFormat.REWARDED;
    }

    @Override
    protected int getAdFormatStringId() {
        return R.string.rewarded_ad_format;
    }

    @Override
    protected void loadAd() {
        assert mPlacementId != null;
        mLogView.print("Loading ...");
        RewardedAd.load(getContext(), mPlacementId, new RewardedAdLoadListener() {
            @Override
            public void onAdLoaded(@NonNull RewardedAd ad) {
                mRewardedAd = ad;
                mLogView.print("Loaded");
                enableShowAd();
                enableDestroyAd();
            }

            @Override
            public void onAdFailedToLoad(@NonNull AdError error) {
                mLogView.print(error.getMessage());
            }
        });
    }

    @Override
    protected void showAd() {
        if (mRewardedAd != null) {
            if (mRewardedAd.isAdInvalidated()) {
                mLogView.print("Ad is invalidated.");
                destroyAd();
                return;
            }
            mRewardedAd.show(requireActivity(), new RewardedAdInteractionListener() {
                @Override
                public void onAdClicked() {
                    mLogView.print("Clicked!");
                }

                @Override
                public void onAdDisplayed() {
                    mLogView.print("Displayed!");
                }

                @Override
                public void onAdDismissed() {
                    mLogView.print("Dismissed");
                    destroyAd();
                }

                @Override
                public void onAdFailedToShow(@NonNull AdError error) {
                    mLogView.print(error.getMessage());
                }

                @Override
                public void onUserRewarded(@NonNull RewardItem reward) {
                    mLogView.print("rewarded: type=" + reward.type + ", amount=" + reward.amount);
                }
            });
            enableDestroyAd();
            disableShowAd();
        }
    }

    @Override
    protected void destroyAd() {
        super.destroyAd();
        mRewardedAd.destroy();
        mRewardedAd = null;
        disableDestroyAd();
        disableShowAd();
    }
}

File: Sample-kotlin/src/main/kotlin/com/opera/ads/demo/fragment/RewardedFragment.kt
package com.opera.ads.demo.fragment

import com.opera.ads.AdError
import com.opera.ads.AdFormat
import com.opera.ads.RewardItem
import com.opera.ads.RewardSsvOptions
import com.opera.ads.demo.R
import com.opera.ads.rewarded.RewardedAd
import com.opera.ads.rewarded.RewardedAd.Companion.load
import com.opera.ads.rewarded.RewardedAdInteractionListener
import com.opera.ads.rewarded.RewardedAdLoadListener

class RewardedFragment : BaseFragment(AdFormat.REWARDED, true, true, R.string.rewarded_ad_format) {
    private var mRewardedAd: RewardedAd? = null

    override fun loadAd() {
        checkNotNull(placementId)
        logger.print("Loading ...")
        load(requireContext(), placementId!!, object : RewardedAdLoadListener {
            override fun onAdLoaded(ad: RewardedAd) {
                mRewardedAd = ad
                logger.print("Loaded")
                enableShowAd()
                enableDestroyAd()
            }

            override fun onAdFailedToLoad(error: AdError) {
                logger.print(error.message)
            }
        })
    }

    override fun showAd() {
        mRewardedAd?.run {
            if (isAdInvalidated()) {
                logger.print("Ad is invalidated.")
                destroyAd()
                return
            }
            // Set scene ID and SSV options(if any) when S2S callback is set for Opera placement id.
            // scene id: max length 100 bytes after url encoded, or will be discarded.
            setSceneId("Demo scene #1")
            setRewardSsvOptions(
                RewardSsvOptions.Builder()
                    // user id: max length 100 bytes after url encoded, or will be discarded.
                    .userId("Demo user id %:{测试?}")
                    // custom data: max length 1KB after url encoded, or will be discarded.
                    .customData("Demo user custom data %:{测试?}#1")
                    .build()
            )
            show(requireActivity(), object : RewardedAdInteractionListener {
                override fun onAdClicked() {
                    logger.print("Clicked!")
                }

                override fun onAdDisplayed() {
                    logger.print("Displayed!")
                }

                override fun onAdDismissed() {
                    logger.print("Dismissed")
                    destroyAd()
                }

                override fun onAdFailedToShow(error: AdError) {
                    logger.print(error.message)
                }

                override fun onUserRewarded(reward: RewardItem) {
                    logger.print("rewarded: type=" + reward.type + ", amount=" + reward.amount)
                }
            })
            enableDestroyAd()
            disableShowAd()
        }
    }

    override fun destroyAd() {
        mRewardedAd?.destroy()
        mRewardedAd = null
        logger.print("Destroyed...")
        disableDestroyAd()
        disableShowAd()
    }
}
